### ISSUE TITLE
Hand embroidery

### ISSUE THREAD

**Issue by fugueink** [inkstitch/inkstitch#1827]
Hi, all!

I know this is something that should be on the Facebook group, but . . . I cannot use Facebook.

Yes, I've tried to use Facebook. I'm autistic, and those pages are just too complicated (and animated!) for me. I just can't cope.

So I can't ask if it's possible to work backwards from a machine embroidery file to make it back into a plain SVG, with objects and solid fills and other things that would allow me to turn it into a hand embroidery design.

I would NOT be distributing the results. I just don't have an embroidery machine, Urban Threads is deleting all their hand designs, and I do not have the money to buy even a few of them before that happens. So I want to know if I can relatively easily  (not DAYS per design) work back to a simple outline SVG image.

I don't need to know how to do it just yet. I just want to know if Inkstitch would be a dead end for my purposes or not.

I am on macOS High Sierra 10.13.6, Inkscape 1.2.1. No, I cannot upgrade to a more recent Macintosh system.

Thank you! My apologies for this use of GitHub!

Edit: Corrected the spelling of "recent" (I used to be a copy editor, so I could not let it stand). I suppose it's a Freudian slip:  I resent that I can't buy a newer machine that runs a newer system!

**Comment by tyrosinase** [comment]
Eh, the Facebook group is just a fan group, where this is the official group. We kinda use it as a forum already, it's fine. (There's Discussions as well but really, we have more conversations over here because those don't notify the same for some people apparently.)

At any rate, it's sorta possible. I've done it some, because I like to kitbash and resize designs, send them to the Silhouette, whatever.

If you're hand-embroidering, maybe you just want to import it and print it, in which case Ink/Stitch is fine. It can print the thing as a realistic version or as an Inkscape graphic... clear or hide the trim marks and it just displays as a bunch of lines - literally it imports designs as Inkscape lines with straight segments where the nodes are the needle-downs.

If you want the solid fills to actually be solid, that's a little more complicated. You can change fill stitches back into fill objects somewhat crudely, same way as your add knockdowns https://silverseams.com/2020/09/knockdown-stitching or dielines https://silverseams.com/2020/08/adding-die-lines-to-embroidery-designs-with-ink-stitch/ except you aren't wanting the fill to expand at all. This works less well on loose fills, of course, but it's pretty close on a lot of designs.

Running stitches are running stitches; like I said it's just a bunch of Inkscape lines. Bean stitches do a lot of back-and-forth and I've noodled around with writing a little Perl script that removes all the duplicate points on the line, with mixed success.

Satins likewise; the Perl script can kinda detect that it's going back and forth and split the two columns back into two columns, though of course all the lock stitches and underlay confuse it, but once again you're just going to see it as a satin stitch so that might be fine.

So allllll that said, if I go look at UT20042 versus UTH20040, this is what it looks like imported.
![2022-09-17_15-19](https://user-images.githubusercontent.com/9473609/190873235-7cfa91fd-d96e-4683-9e42-f590f803c070.png)

Troubleshooter will remove the trims, or you can just hide them. The machine design has two colors of shading in back of the line art, which is pretty easy to hide.
![2022-09-17_15-20](https://user-images.githubusercontent.com/9473609/190873265-0d2645bb-7062-4981-a87e-ce61119149f6.png)

That leaves the satin stitch, which looks like is just done as an outline in the hand-stitching version. It'll end up looking like this.
![2022-09-17_15-21](https://user-images.githubusercontent.com/9473609/190873310-0a389dde-0d08-4039-9b28-b34490c15cd9.png)

(Truly, I have some questions about UT's software, the edges of that satin are *drunk*)

I don't know what the hand-stitching ones look like as far as telling you what areas to stitch first, but I can tell you the machine stitching version will be useless for that. It stitches the dragon as a single continuous path, and it's absolutely mesmerizing to watch (like a Roomba!) because you never know what choice it's going to make at any intersection. It may lay down a single line of stitching somewhere and wander away and lose interest and finally come back and remember to go over it a couple more times a couple thousand stitches later.

**Comment by fugueink** [comment]
Sigh. What I am doing looks like mostly satin stitch.

![piggy](https://user-images.githubusercontent.com/1401986/190874573-70d16c5e-a761-472e-93fd-9802f10b523d.png)

I'm pretty much looking to get something that looks like the simple line version of the actual dragon. I can look at the sample image on the CD cover to see what color goes where. And I can look at the original file to see what direction I should make my stitches go.

Any suggestions on how to get the edges of the satin stitch into a line?

Thank you!

**Comment by fugueink** [comment]
I've continued to poke around.

I think I can do it, _if_ I can find some way to shrink those nodes back down to the usual size for nodes outside of InkStitch.

![GinormousNodes!](https://user-images.githubusercontent.com/1401986/190877832-0995d76f-3b9a-4bdd-bf8b-241410b9ae03.png)

Anyone got any ideas for that? I've searched for changing the size of the node indicators in both InkStitch and InkScape, and I cannot find it for either one.

Thanks!
